core/app_streak_notifications.go

```diff
@@ -0,0 +1,103 @@
+/*
+ *   Copyright (c) 2025 Board of Trustees of the University of Illinois.
+ *   All rights reserved.
+
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package core
+
+import (
+	"application/core/interfaces"
+	"application/core/model"
+	"application/driven/notifications"
+	"application/utils"
+	"fmt"
+	"time"
+
+	"github.com/rokwire/rokwire-building-block-sdk-go/utils/logging/logs"
+)
+
+type streakNotifications struct {
+	logger *logs.Logger
+
+	notifications interfaces.Notifications
+
+	storage interfaces.Storage
+
+	//streak notifications timer
+	streakNotificationsTimer     *time.Timer
+	streakNotificationsTimerDone chan bool
+}
+
+func (n streakNotifications) start() {
+	//setup notifications timer
+	go n.setupStreakNotificationsTimer()
+}
+
+func (n streakNotifications) setupStreakNotificationsTimer() {
+	//cancel if active
+	if n.streakNotificationsTimer != nil {
+		n.logger.Info("setupStreakNotificationsTimer -> there is active timer, so cancel it")
+
+		n.streakNotificationsTimerDone <- true
+		n.streakNotificationsTimer.Stop()
+	}
+
+	now := time.Now().UTC()
+	nowSecondsInDay := utils.SecondsInHour*now.Hour() + utils.SecondsInMinute*now.Minute() + now.Second()
+	desiredMoment := 0 //default desired moment of the day in seconds (beginning of the day)
+
+	var durationInSeconds int
+	n.logger.Infof("setupStreakNotificationsTimer -> nowSecondsInDay:%d", nowSecondsInDay)
+	if nowSecondsInDay <= desiredMoment {
+		n.logger.Info("setupStreakNotificationsTimer -> notifications not yet processed today")
+		durationInSeconds = desiredMoment - nowSecondsInDay
+	} else {
+		n.logger.Info("setupStreakNotificationsTimer -> notifications have already been processed today")
+		durationInSeconds = (utils.SecondsInDay - nowSecondsInDay) + desiredMoment // the time which left today + desired moment from next day
+	}
+
+	initialDuration := time.Second * time.Duration(durationInSeconds)
+	utils.StartTimer(n.streakNotificationsTimer, n.streakNotificationsTimerDone, &initialDuration, time.Duration(utils.HoursInDay)*time.Hour, n.processNotifications, "processStreakNotifications", n.logger)
```

Comment:
We do have multiple instances of the surveys BB running so I think we need to add some handling that prevents two instances from sending the same notifications. I assume this will need to involve storing a config record in the database.